### ISSUE TITLE
2.0rc2:  title too close to frame?

### ISSUE THREAD

**Issue by stggh** [matplotlib/matplotlib#7660]
Matplotlib version 2.0rc2
installed via `conda install -c conda-forge/label/rc -c conda-forge matplotlib`

As per the image below, a subscript `g` is truncated when combined with a superscript. In the example, the Greek Sigma may be any other letter:

```python
import matplotlib.pyplot as plt

plt.title(r"superscript '-' truncates subscript 'g'        $\Sigma_{g}^{-}$")

plt.savefig("truncated-title-subscript.png", dpi=75)
plt.show()
```

![truncated-title-subscript](https://cloud.githubusercontent.com/assets/10932229/21412677/2ae56db8-c846-11e6-8446-dca357ca9cda.png)

**Comment by robbmcleod** [comment]
Probably a related issue, it also crops the x- and y-axes labels.

![opt7_width10_correlatednoise_vs_cumdose](https://cloud.githubusercontent.com/assets/5999769/21420408/e38bafee-c82d-11e6-8fef-f56102c7d0f0.png)

This is for a fontsize of 18, with 14 it's almost okay.  

plt.rc( 'lines', linewidth=2 )
plt.rc( 'font', size=18 )
plt.rc( 'font', family='serif' )
plt.rc( 'font', serif='Times New Roman' )

**Comment by tacaswell** [comment]
For the first issue I think we can adjust the default padding a bit.

For @robbmcleod there is not much we can do about that without a 'proper' layout manager.  See http://matplotlib.org/2.0.0rc2/users/dflt_style_changes.html#figure-size-font-size-and-screen-dpi for more details.  The default paddings / axes size / figure size are tuned for the default font size.  If you want to change the default font size you will have to also change the others.

attn @efiring

**Comment by stggh** [comment]
Thanks.

Sorry to add noise to this discussion, but I have always wondered, is there a reason why the annotation font size does not scale with the image zoom?

**Comment by tacaswell** [comment]
That decision was made long ago, but I suspect it is to make it easy to match the font size of any text to the font of the document you are embedding the figure in.  

Font size specified in dataspace would be weird to work with (and what should it do with non-linear scales?).

**Comment by efiring** [comment]
Also, the point of zooming is to see more detail in the data, not to see everything blown up uniformly.

**Comment by stggh** [comment]
All a matter of preference:

- I find that if the image size is changed, depending on use, publication vs presentation, then all the annotation must be resized and relocated. 

- If a particular (spectral) label needs to be carefully positioned between two peaks, zooming in does not assist with the location, because the text font size reduces.

But, I now note, that some other graphics packages also do not zoom scale fonts, so matplotlib is consistent.

Thanks for all of your magnificent work.

**Comment by efiring** [comment]
Regarding the original problem pointed out here: I think I increased the title default font size without increasing its pad.  Because the default vertical alignment is 'baseline', the descenders are nearly touching the Axes outline, and then anything beyond, like a subscript, goes over the edge.  For v2.0, we can increase the pad, which I think will look better even with normal text.  A large increase is needed (from 4 to 6) to handle the example given here.

An alternative approach is to use the bottom of the text rather than the baseline; the problem is that this works badly if one uses more than one title (e.g., a left and a right) and they have different extents below the baseline.  (Passing  `fontdict=dict(verticalalignment='bottom')` in the `set_title` call achieves this.)

A better API might include either or both of 2 changes: 
- Express the pad (always, or perhaps optionally) in terms of the font size, so that it scales with the font, instead of in points.
- Add a 'pad' kwarg to the `set_title` method.  At present, the only way to change it is via the rcParam, which means it can't be changed without completely clearing the axes.

Both of these things are doable, but neither is quite a simple as one might expect.

**Comment by efiring** [comment]
Closed by 276c489a.
